runtime/criusupport/criusupport.cpp

```diff
@@ -592,6 +622,13 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 			/* throw the pending exception */
 			goto wakeJavaThreads;
 		}
+
+		if (vm->checkpointState.isJdwpEnabled) {
+			acquireSafeOrExcusiveVMAccess(currentThread, vmFuncs, safePoint);
```

Comment:
this is done too early, shutting down GC threads could trigger a JVMTI event, if all JVMTI threads are blocked then this will lead to a deadlock. This happens fairly consistenly on my machine.

The JVMTI threads need to be shutdown after `runInternalJVMCheckpointHooks` is called

Reply:
Moved the disabling of JDWP threads after the internal hooks before checkpoint, and re-enabling of JDWP threads to before the GC reinitializes threads after restore.